src/software/world/ball.h

```diff
@@ -61,13 +66,29 @@ class Ball final
      */
     Timestamp lastUpdateTimestamp() const;
 
+    /**
+     * Gets the previous timestamps for each state stored in states_
+     *
+     * @return Vector containing the update timestamp history starting with the oldest
+     * available data at index 0
+     */
+    std::vector<Timestamp> getPreviousTimestamps() const;
+
     /**
      * Returns the current position of the ball
      *
      * @return the current position of the ball
      */
     Point position() const;
 
+    /**
+     * Gets the previous positions for each state stored in states_
+     *
+     * @return Vector containing the position history starting with the oldest available
+     * data at index 0
```

Comment:
Isn't the newest data at the "front" of the buffer which is index 0?

Reply:
The old class was using push_back so I used it and I think it loops from front to back so it'll be the oldest one